improver_tests/lapse_rate/test_LapseRate.py

```diff
@@ -108,6 +108,25 @@ def test_returns_expected_values(self):
             self.orography)
         self.assertArrayAlmostEqual(result, expected_out)
 
+    def test_returns_expected_values_generator(self):
+        """Test that the function returns True at points where the height
+           difference to the central pixel is greater than 35m."""
+
+        expected_out = np.array(
+            [[True, True, False, False, False, False, False, False, True],
+             [True, True, False, False, False, False, False, False, True]])
+
+        result = np.zeros(expected_out.shape, dtype=bool)
+        gen = LapseRate(nbhood_radius=1)._generate_heightdiff_mask(
+            self.orography)
+
+        count = 0
```

Comment:
Suggest avoiding manual handling of counter variable - similar to other comment above.

Reply:
Agreed, I'll fix it now.